jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/conf/JdbcStorageConfigManager.java

```diff
@@ -94,4 +113,57 @@ private static boolean isEmptyString(String value) {
     return ((value == null) || (value.trim().isEmpty()));
   }
 
+  private static void resolveMetadata(Properties props) {
+    try {
+      DatabaseType dbType = DatabaseType.valueOf(
+        props.getProperty(JdbcStorageConfig.DATABASE_TYPE.getPropertyName()));
+
+      LOGGER.debug("Resolving db type: {}", dbType.toString());
+
+      if (dbType == DatabaseType.METASTORE) {
+        HiveConf hconf = Hive.get().getConf();
+        props.setProperty(JdbcStorageConfig.JDBC_URL.getPropertyName(),
+            getMetastoreConnectionURL(hconf));
+        props.setProperty(JdbcStorageConfig.JDBC_DRIVER_CLASS.getPropertyName(),
+            getMetastoreDriver(hconf));
+
+        String user = getMetastoreJdbcUser(hconf);
+        if (user != null) {
+          props.setProperty("hive.sql.dbcp.username", user);
```

Comment:
The properties here are being added to the configuration. All "dbcp" properties will be used by the dbcp. See GenericJdbcDatabaseAccessor.